### PR TITLE
fix: builder fails on intended JSON

### PR DESCRIPTION
**Pull request by matyasfodor**
I think the most robust solution is to use base64 encoding. I was also thinking about doing the same for [json_dump_and_escape](https://github.com/zakandrewking/escher/blob/master/py/escher/escape.py#L16).

It's meant to fix https://github.com/zakandrewking/escher/issues/229

### CODE REVIEW

File: py/escher/escape.py [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+    return json_string.replace('\\', '\\\\').replace("'", "\\'").replace("\n", "\\n")
 
 def escape_json_or_null(json_string):
     """Escape the JSON string. If None, then return JSON 'null'."""
     if json_string is None:
-        return json.dumps(None)
-    else:
-        return _escape_for_embedding(json_string)
+        json_string = json.dumps(None)
+    return base64.b64encode(json_string.encode('ascii')).decode('ascii')
```
Comment (zakandrewking) [inline py/escher/escape.py]: wouldn't the first encode need to be UTF8, since JSON is always UTF8?

File: py/escher/escape.py [hunk 2]
```diff
@@ -1,17 +1,17 @@
 # -*- coding: utf-8 -*-
 
 import json
+import base64
 
 
 def _escape_for_embedding(json_string):
-    return json_string.replace('\\', '\\\\').replace("'", "\\'")
+    return json_string.replace('\\', '\\\\').replace("'", "\\'").replace("\n", "\\n")
```
Comment (zakandrewking) [inline py/escher/escape.py]: will we need to escape at all now? the base64 encoding can replace this, right?

File: py/escher/escape.py (hunk 1 above)
Comment (zakandrewking) [reply py/escher/escape.py]: it would be good to have a test case for this

**Comment by matyasfodor** [comment]
@zakandrewking I updated the code and I decided to move out the converter function from escape to util since it's just a couple of lines. I found a few problems though:
 - I had to delete the test that checks that builder options are changed properly in the rendered HTML.
 - I think for urls there's no need for JSON conversion, but I updated that part as well.
 - Turns out js `atob` doesn't support utf-8, so I'll have to add an npm package or a few lines of conversion code: https://stackoverflow.com/a/30106551/2419215

**Comment by matyasfodor** [comment]
@zakandrewking I made the changes, but I think github won't notice it since I squashed my commits. Could you please review it again?

**Comment by zakandrewking** [comment]
I tried a basic session in Jupyter:

```
import escher
b = escher.Builder(map_name='e_coli_core.Core metabolism')
b.display_in_notebook()
```

And it crashed with this error:

```
Javascript error adding output!
SyntaxError: Unexpected token o in JSON at position 0
See your browser Javascript console for more details.
```

I'm having a hard time debugging this at the moment. Any idea what could be happening?

**Comment by zakandrewking** [comment]
Local development is a little tricky, unfortunately.

If you want to try it, the steps are:

```bash
yarn build && yarn copy
cd py
pip install -e .
python -m escher.server # keep this server running
```

Then, when you create a map in Jupyter:

```python
b = escher.Builder(local_host='http://localhost:7778/')
b.display_in_notebook(js_source='local')
```

I'd like to make this process easier when we find time to refactor the Python codebase.

In the meantime, I checked you PR, and it looks good. Merging!

**Pull request merged**